drivers/power/pm_register.c

```diff
@@ -56,20 +56,24 @@
 
 int pm_register(FAR struct pm_callback_s *callbacks)
 {
-  int ret;
-
   DEBUGASSERT(callbacks);
 
   /* Add the new entry to the end of the list of registered callbacks */
 
-  ret = pm_lock();
-  if (ret == OK)
+  if (OSINIT_OS_READY())
```

Comment:
I think this completely contradicts to https://github.com/apache/incubator-nuttx/pull/5849
@GUIDINGLI could you please review this change?